saleor/graphql/product/mutations/product_variant/product_variant_update.py

```diff
@@ -162,11 +200,28 @@ def perform_mutation(  # type: ignore[override]
             "external_reference",
             external_reference,
         )
-        return super().perform_mutation(
-            root,
-            info,
-            external_reference=external_reference,
-            id=id,
-            sku=sku,
-            input=input,
+        instance = cls.get_instance(
+            info, id=id, sku=sku, external_reference=external_reference, input=input
         )
+        old_instance_data = model_to_dict(instance).copy()
+
+        cleaned_input = cls.clean_input(info, instance, input)
+        metadata_list = cleaned_input.pop("metadata", None)
+        private_metadata_list = cleaned_input.pop("private_metadata", None)
+        instance = cls.construct_instance(instance, cleaned_input)
```

Comment:
If the approach from this PR is "common" thing that we want to introduce, then I would consider extending ModelMutation to have something like :
```python
def update_instance(cls, instance, cleaned_data)-> tuple[Object, list[CHANGED_FIELD_NAME]]:
```
Then as a result you could get the `instance` and fields that have been changed (you could use similar logic from `construct_instance` but with additional tracking of changed field names). 
Then instead of passing `base_fields_changed` to the `_save`, you could pass these modified fields.
Then in `_save`
```python
@classmethod
def _save(cls, info: ResolveInfo, instance, cleaned_input, modifed_field_names):
    ...
    if modifed_field_names:
        instance.save(updated_fields=modifed_field_names)
```
So instead of making whole `save` you will call save only on modified fields. Additionally, introducing similar approach for mutations like ProductUpdated etc should be easier

Reply:
Yes, I will do so, but in separate PR, for now we just want to quickly optimize case "If nothing changed, do not save at all".

Reply:
Sure, sounds good 👍